src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/Program.cs

```diff
@@ -21,57 +133,101 @@ public void Main(string[] args)
+                    }
+                    break;
+                case "DuplexStreaming":
+                    if (_paramUseAsync)
+                    {
+                        DoTheRunAsync(RunAllStressTestsAsync<IDuplexStreamingService, DuplexStreamingTest<IDuplexStreamingService, DuplexStreamingStressTestParams>, DuplexStreamingStressTestParams>);
+                    }
+                    else
+                    {
+                        DoTheRun(RunAllStressTests<IDuplexStreamingService, DuplexStreamingTest<IDuplexStreamingService, DuplexStreamingStressTestParams>, DuplexStreamingStressTestParams>);
+                    }
+                    break;
+                default:
+                    Console.WriteLine("wrong argument: " + _paramTestToRun);
+                    return;
             }
-            Console.WriteLine("hostName: " + hostName);
+        }
 
-            TestHelpers.SetHostAndProtocol(useHttp: false, hostName: hostName, appName: "WcfService1");
+        private void DoTheRun(Action test)
+        {
+            DoTheRunAsync(() => { test(); return Task.FromResult(true); });
         }
 
-        private static void DoTheRun(int iterations)
+        private void DoTheRunAsync(Func<Task> testAsync)
         {
+            var cts = new CancellationTokenSource(_paramStressRunDuration);
             Console.WriteLine("Start");
-            Task[] allTasks = new Task[13];
+
+            Task[] allTasks = new Task[_paramStressLevel];
             for (int t = 0; t < allTasks.Length; t++)
             {
                 int tt = t;
-                allTasks[t] = Task.Run(() =>
+                allTasks[t] = Task.Run(async () =>
                 {
                     var ttt = tt;
-                    for (int i = 0; i < iterations; i++)
+                    for (long i = 0; i < _paramIterations / allTasks.Length; i++)
                     {
                         try
                         {
-                            //RunAllHttpStressTests();    //  for TestHelpers.SetHostAndProtocol(useHttp: true
-                            RunAllNetTcpStressTests();  //  for TestHelpers.SetHostAndProtocol(useHttp: false
-
+                            await testAsync();
+                        }
+                        catch (ObjectDisposedException e)
+                        {
+                            TestUtils.ReportFailure(e.ToString());
                         }
                         catch (Exception e)
                         {
-                            System.Diagnostics.Debugger.Break();
+                            TestUtils.ReportFailure(e.ToString());
+                            throw;
                         }
 
-                        if (i % 1000 == 0)
+                        if (i % 100 == 0)
                         {
-                            Console.WriteLine(ttt + " " + i);
+                            Console.WriteLine(DateTime.Now.ToString() + " " + ttt + " " + i);
```

Comment:
This method isn't used when running perf (and no threads that actually make requests in perf write to Console). Perhaps I should rename the method to avoid confusion.

However your suggestion to use a different logging mechanism still makes perfect sense.